AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/hierarchical-state-machine-swift/StateMachine.swift

```diff
@@ -220,13 +163,11 @@ extension StateMachine: EventDispatcher {
     /// machine's state and invokes listeners with the new state. Regardless of whether
     /// the state is new or not, the state machine will execute any effects from the
     /// event resolution process.
-    func send(_ event: StateMachineEvent) {
-        operationQueue.addOperation {
-            self.process(event: event)
-        }
+    func send(_ event: StateMachineEvent) async {
+        process(event: event)
```

Comment:
New behavior note: `send` will now always invoke state machine change listeners before it returns. Callers should be ready to handle state machine change updates before invoking `send`. Previously, the order of "return from method" vs. "invoke change listener" was undefined, but tended to be that it returned first.